### PR TITLE
UI: Disallow exiting settings with no audio tracks selected

### LINKED ISSUE #8887: Not selecting an audio track causes output to error
### Operating System Info

Windows 11

### Other OS

_No response_

### OBS Studio Version

29.1.1

### OBS Studio Version (Other)

_No response_

### OBS Studio Log URL

https://obsproject.com/logs/b5fdYPrxHOng1ZhQ

### OBS Studio Crash Log URL

_No response_

### Expected Behavior

I don't think it should be possible to select no audio tracks if it will always error. Also, this error is not logged. Advanced output shows a "You must select at least one track" error in the UI while simple mode does not.

### Current Behavior
image

image

### Steps to Reproduce

1. Select simple output mode
2. Set recording quality to anything besides Same as Stream
3. Try to start your recording
...

### Anything else we should know?

_No response_

### PR DESCRIPTION
**Pull request by gxalpha**
### Description

Disallows the user from exiting the audio settings without having selected an audio track.

image

Split into two commits so that the first commit (for simple mode) can be cherry-picked to the 29.1 branch, while the second commit can wait until the next version.

### Motivation and Context

Fixes #8887

### How Has This Been Tested?

macOS 13.3.1
Unselected all audio tracks in both simple and advanced mode, and saw the popup appear, preventing me from exiting settings. Selected an audio track and exited successfully.
Selected FLV (and "Same as Stream" quality in simple mode) with none of the checkboxes checked beforehand and exited the settings successfully (since the checkboxes don't have an effect with FLV or with Same as stream quality in simple mode).

### Types of changes

- Bug fix (non-breaking change which fixes an issue)

- Tweak (non-breaking change to improve existing functionality)

### Checklist:

- [x] My code has been run through [clang-format](https://github.com/obsproject/obs-studio/blob/master/.clang-format).
- [x] I have read the [**contributing** document](https://github.com/obsproject/obs-studio/blob/master/CONTRIBUTING.rst).
- [x] My code is not on the master branch.
- [x] The code has been tested.
- [x] All commit messages are properly formatted and commits squashed where appropriate.
- [x] I have included updates to all appropriate documentation.

### CODE REVIEW

File: UI/data/locale/en-US.ini [hunk 1]
```diff
@@ -1331,6 +1331,7 @@ SceneItemShow="Show '%1'"
 SceneItemHide="Hide '%1'"
 
 # Output warnings
+OutputWarnings.NoTracksSelected.Title="No audio track selected"
```
Comment (RytoEX) [inline UI/data/locale/en-US.ini:1334]:
```suggestion
OutputWarnings.NoTracksSelected.Title="No Audio Track Selected"
```

File: UI/window-basic-main-outputs.cpp [hunk 2]
```diff
@@ -1786,6 +1786,10 @@ inline void AdvancedOutput::SetupRecording()
 	unsigned int cy = 0;
 	int idx = 0;
 
+	/* Hack to allow recordings without any audio tracks selected. It is no
+	 * longer possible to select such a configuration in settings, but legacy
+	 * configurations might still have this configured and we don't want to
+	 * just break them. */
```
Comment (RytoEX) [inline UI/window-basic-main-outputs.cpp:1792]: How long has this been possible? Wondering if there's a way for us to force the user to select a track by preventing an output from starting in this configuration, as a future approach.

**Review by RytoEX — CHANGES_REQUESTED** [review]

File: UI/data/locale/en-US.ini (hunk 1 above)
Comment (gxalpha) [reply UI/data/locale/en-US.ini:1334]: Fixed

**Comment by Warchamp7** [comment]
image

Title -> `Output Settings Error`
Text -> `All outputs must have at least one audio track selected`

**Comment by gxalpha** [comment]
Adjusted the wording per the above comment.
image

**Review by Warchamp7 — APPROVED** [review]

**Review by RytoEX — APPROVED** [review]
After some off-thread discussion, this is better than the current situation. It might be ideal if the user selects zero audio tracks, that OBS outputs zero audio tracks if OBS can handle that, but for the interim, this is better than causing outputs to fail.

**Comment by jgusta** [comment]
## Paradox 
In the event that you have all audio tracks unselected and all audio devices turned off, you enter a Catch-22 situation where you can't turn the audio device on because you can't hit 'Apply' without selecting audio tracks. You can't select audio tracks unless you have an audio device selected.

### Device disabled:
CleanShot 2023-08-18 at 18 27 03@2x

### Enabled, hitting apply:
CleanShot 2023-08-18 at 18 27 16@2x

### Can't select audio track
CleanShot 2023-08-18 at 18 36 26@2x

### Can't select audio track in recording settings
CleanShot 2023-08-18 at 18 22 55@2x

### Can't select bitrate for audio tracks
CleanShot 2023-08-18 at 18 24 06@2x

**Comment by gxalpha** [comment]
@jgusta on first glance it looks like the track selection is disabled because there’s an output running. Try stopping any recording/stream/virtualcam/NDI/etc, and then try to select a recording track. It should have nothing to do with whether there are any input devices selected.
If that doesn’t fix it, please file a new issue, reports under merged PRs can get lost easily. I’m currently on vacation but would assign it to me and look at it in detail once I’m back.

**Comment by jgusta** [comment]
I've got it fixed. Apologies for posting on this PR. But to wrap up, it was a scene with the NDI plugin turned on that triggered the output error. Thank you.

**Comment by iG8R** [comment]
Seams the issue still exists.
 - Microsoft Windows [Version 10.0.19045.2673] 64bit
 - OBS ver 30.0
 
image
image

**Pull request merged**